### PR TITLE
CI: Update to Pyodide 0.23.1

### PR DESCRIPTION
**Pull request by hoodmane**

### CODE REVIEW

**Comment by hoodmane** [comment]
Uh oh seems like there may be a regression @ryanking13 I think the xbuildenv is missing `python_stdlib.zip`...

**Comment by charris** [comment]
@hoodmane Any progress on this? Or can we close it.

**Comment by hoodmane** [comment]
Yes I know how to fix it will come back to it hopefully in the next few days.

**Comment by hoodmane** [comment]
Well not this week but hopefully next.

**Comment by hoodmane** [comment]
Well I think all that needs to happen is to bump the version of Pyodide to 0.23.1

File: .github/workflows/emscripten.yml [hunk 1]
```diff
@@ -21,13 +21,13 @@ jobs:
     runs-on: ubuntu-22.04
     if: "github.repository == 'numpy/numpy'"
     env:
-      PYODIDE_VERSION: 0.22.1
+      PYODIDE_VERSION: 0.23.0
```
Comment (seberg) [inline .github/workflows/emscripten.yml:24]:
```suggestion
      PYODIDE_VERSION: 0.23.1
```
I guess this is probably not all that is needed?  Let's just try :)

**Comment by seberg** [comment]
Well, Emscripten CI passed, so putting this in.  Thanks @hoodmane please just follow up/make a note if bumping to piodide 0.23.1 wasn't the whole story.

**Comment by hoodmane** [comment]
I think that's it. Thanks @seberg!

**Pull request merged**